### PR TITLE
Replaced all standard log entries with klog Fixes #324

### LINKED ISSUE #324: Replace klog in the onos-config project with either standard log package or something better
onos-config and its associated client programs use a combination of the following, each with their own benefits and drawbacks:

- klog (requires some k8s dependencies. Output is not seen on stdout or stderr. Log levels are configurable)

- standard log (from https://golang.org/pkg/log/, so no extra dependencies. Does not have levels of logging other than Print of Fatal. Works with k8s and stdout and stderr. Doesn't colour red for error)

- glog (from https://github.com/golang/glog. Has Info, Warning etc. Prints to stdout and stderr and works with k8s)

- fmt.println (standard go package. Should only be used in client programs for output to stdout

We should try to harmonize on one approach for logging, and make sure fmt.print line follows the given approach

### PR DESCRIPTION
**Pull request by SeanCondon**
klog.SetOutput(os.Stdout) is called in

- onos-config - should it always be called in here - will it impede writing to K8s logs?
- each TestMain

fmt.Println is still left there for Command Line utils and their associated pkg files

### CODE REVIEW

File: pkg/dispatcher/dispatcher.go [hunk 1]
```diff
@@ -51,15 +51,15 @@ func NewDispatcher() Dispatcher {
...
-	log.Println("Event listener initialized")
+	log.Info("Event listener initialized")
 
 	for configEvent := range changeChannel {
-		log.Println("Listener: Event", configEvent)
+		log.Info("Listener: Event", configEvent)
 		deviceChan, ok := d.deviceListeners[topocache.ID(events.Event(configEvent).Subject())]
 		if ok {
-			log.Println("Device Simulators must be active")
+			log.Info("Device not connected - config event discarded", configEvent)
```
Comment (Andrea-Campanella) [inline pkg/dispatcher/dispatcher.go]: this probably ought to be at least a warning

File: pkg/dispatcher/dispatcher.go [hunk 2]
```diff
@@ -51,15 +51,15 @@ func NewDispatcher() Dispatcher {
...
 			//TODO need a timeout or be done in separate routine, blocks if there is some error underneath
-			log.Println(deviceChan)
+			log.Info(deviceChan)
```
Comment (Andrea-Campanella) [inline pkg/dispatcher/dispatcher.go]: this can be removed

File: pkg/manager/manager_test.go [hunk 3]
```diff
@@ -114,7 +115,7 @@ func setUp() (*Manager, map[string]*change.Change, map[store.ConfigName]store.Co
 		},
 		make(chan events.TopoEvent, 10))
 	if err != nil {
-		fmt.Println(err)
+		log.Fatal(err)
```
Comment (Andrea-Campanella) [inline pkg/manager/manager_test.go]: is this fatal preventing the -1 exit to the os effectively blocking the test and not failing it ?

File: pkg/northbound/gnmi/subscribe.go [hunk 4]
```diff
@@ -89,12 +89,12 @@ func listenOnChannel(stream gnmi.GNMI_SubscribeServer, mgr *manager.Manager, has
...
 			code, ok := status.FromError(err)
 			if ok && code.Code() == codes.Canceled {
-				log.Println("Subscription Terminated, Canceled")
+				log.Info("Subscription Terminated, Canceled")
 				mgr.Dispatcher.UnregisterNbi(hash)
 				mgr.Dispatcher.UnregisterOperationalState(hash)
 				resChan <- result{success: true, err: nil}
 			} else {
-				log.Println("Error in subscription", err)
+				log.Info("Error in subscription", err)
```
Comment (Andrea-Campanella) [inline pkg/northbound/gnmi/subscribe.go]: Error I would say

File: pkg/southbound/clientManager.go [hunk 5]
```diff
@@ -43,24 +43,24 @@ func createDestination(device topocache.Device) (*client.Destination, DeviceID)
 	d.Target = device.Target
 	d.Timeout = time.Duration(device.Timeout) * time.Second
 	if device.Plain {
-		log.Println("Plain connection connection to", device.Addr)
+		log.Warning("Plain connection connection to", device.Addr)
```
Comment (Andrea-Campanella) [inline pkg/southbound/clientManager.go]: I woudl say info, these only stem from elements that the user gave, thus it's well known

same for line 48

File: pkg/southbound/clientManager.go [hunk 6]
```diff
@@ -43,24 +43,24 @@ func createDestination(device topocache.Device) (*client.Destination, DeviceID)
...
 			d.TLS.RootCAs = getCertPool(device.CaPath)
 		}
 		if device.CertPath == "" && device.KeyPath == "" {
 			// Load default Certificates
-			log.Println("Loading default certificates")
+			log.Info("Loading default certificates")
 			clientCerts, err := tls.X509KeyPair([]byte(certs.DefaultClientCrt), []byte(certs.DefaultClientKey))
 			if err != nil {
-				log.Println("Error loading default certs")
+				log.Info("Error loading default certs")
```
Comment (Andrea-Campanella) [inline pkg/southbound/clientManager.go]: error

File: pkg/manager/setconfig.go [hunk 7]
```diff
@@ -86,16 +86,16 @@ func (m *Manager) SetNetworkConfig(configName store.ConfigName, updates map[stri
 	}
 
 	if m.ChangeStore.Store[store.B64(configChange.ID)] != nil {
-		log.Println("Change ID", store.B64(configChange.ID), "already exists - not overwriting")
+		log.Info("Change ID", store.B64(configChange.ID), "already exists - not overwriting")
```
Comment (Andrea-Campanella) [inline pkg/manager/setconfig.go]: warning ?

File: pkg/manager/setconfig.go [hunk 8]
```diff
@@ -107,23 +107,23 @@ func (m *Manager) SetNetworkConfig(configName store.ConfigName, updates map[stri
...
 	deviceModelYgotPlugin, ok := m.ModelRegistry[modelName]
 	if !ok {
-		log.Println("No model", modelName, "available as a plugin")
+		log.Warning("No model", modelName, "available as a plugin")
 	} else {
 		configValues := deviceConfig.ExtractFullConfig(m.ChangeStore.Store, 0)
 		jsonTree, err := store.BuildTree(configValues)
 		if err != nil {
-			log.Println("Error building JSON tree from Config Values", err, jsonTree)
+			log.Warning("Error building JSON tree from Config Values", err, jsonTree)
```
Comment (Andrea-Campanella) [inline pkg/manager/setconfig.go]: error

File: pkg/manager/setconfig.go [hunk 9]
```diff
@@ -107,23 +107,23 @@ func (m *Manager) SetNetworkConfig(configName store.ConfigName, updates map[stri
...
 		} else {
 			ygotModel, err := deviceModelYgotPlugin.UnmarshalConfigValues(jsonTree)
 			if err != nil {
-				log.Println("Error unmarshaling JSON tree in to YGOT model", err)
+				log.Warning("Error unmarshaling JSON tree in to YGOT model", err)
```
Comment (Andrea-Campanella) [inline pkg/manager/setconfig.go]: error

File: pkg/northbound/gnmi/set.go [hunk 10]
```diff
@@ -173,20 +173,20 @@ func (s *Server) buildUpdateResults(targetUpdates mapTargetUpdates,
...
 			if strings.Contains(err.Error(), manager.SetConfigAlreadyApplied) {
-				log.Println(manager.SetConfigAlreadyApplied, "Change", store.B64(changeID), "to", configName)
+				log.Warning(manager.SetConfigAlreadyApplied, "Change", store.B64(changeID), "to", configName)
 				continue
 			}
 
 			//FIXME this at the moment fails at a device level. we can specify a per path failure
 			// if the store could return us that info
-			log.Println("Error in setting config:", changeID, "for target", configName, err)
+			log.Warning("Error in setting config:", changeID, "for target", configName, err)
```
Comment (Andrea-Campanella) [inline pkg/northbound/gnmi/set.go]: error ?

File: pkg/northbound/gnmi/set.go [hunk 11]
```diff
@@ -173,20 +173,20 @@ func (s *Server) buildUpdateResults(targetUpdates mapTargetUpdates,
...
 			return nil, nil, err
 		}
 
 		for k := range updates {
 			path, errInPath := utils.ParseGNMIElements(strings.Split(k, "/")[1:])
 			if errInPath != nil {
-				log.Println("ERROR: Unable to parse path", k)
+				log.Warning("ERROR: Unable to parse path", k)
```
Comment (Andrea-Campanella) [inline pkg/northbound/gnmi/set.go]: error

File: pkg/northbound/gnmi/set.go [hunk 12]
```diff
@@ -204,7 +204,7 @@ func (s *Server) buildUpdateResults(targetUpdates mapTargetUpdates,
 		for _, r := range targetRemoves[target] {
 			path, errInPath := utils.ParseGNMIElements(strings.Split(r, "/")[1:])
 			if errInPath != nil {
-				log.Println("ERROR: Unable to parse path", r)
+				log.Warning("ERROR: Unable to parse path", r)
```
Comment (Andrea-Campanella) [inline pkg/northbound/gnmi/set.go]: error

File: pkg/northbound/gnmi/subscribe.go [hunk 13]
```diff
@@ -136,7 +136,7 @@ func collector(stream gnmi.GNMI_SubscribeServer, request *gnmi.SubscriptionList,
 		//We get the stated of the device, for each path we build an update and send it out.
 		update, err := getUpdate(request.Prefix, sub.Path)
 		if err != nil {
-			log.Println("Error while collecting data for subscribe once or poll", err)
+			log.Warning("Error while collecting data for subscribe once or poll", err)
```
Comment (Andrea-Campanella) [inline pkg/northbound/gnmi/subscribe.go]: if an error is returned to the resChan please move these to errors

File: pkg/southbound/clientManager.go [hunk 14]
```diff
@@ -43,24 +43,24 @@ func createDestination(device topocache.Device) (*client.Destination, DeviceID)
...
 			d.TLS.RootCAs = getCertPool(device.CaPath)
 		}
 		if device.CertPath == "" && device.KeyPath == "" {
 			// Load default Certificates
-			log.Println("Loading default certificates")
+			log.Info("Loading default certificates")
 			clientCerts, err := tls.X509KeyPair([]byte(certs.DefaultClientCrt), []byte(certs.DefaultClientKey))
 			if err != nil {
-				log.Println("Error loading default certs")
+				log.Warning("Error loading default certs")
```
Comment (Andrea-Campanella) [inline pkg/southbound/clientManager.go]: maybe move to error ?

File: pkg/southbound/clientManager.go [hunk 15]
```diff
@@ -111,7 +111,7 @@ func (target *Target) ConnectTarget(ctx context.Context, device topocache.Device
 func setCertificate(pathCert string, pathKey string) tls.Certificate {
 	certificate, err := tls.LoadX509KeyPair(pathCert, pathKey)
 	if err != nil {
-		log.Println("could not load client key pair", err)
+		log.Warning("could not load client key pair", err)
```
Comment (Andrea-Campanella) [inline pkg/southbound/clientManager.go]: everything related to certs failing to me is an error

File: pkg/southbound/synchronizer/synchronizer.go [hunk 16]
```diff
@@ -52,24 +52,24 @@ func New(context context.Context, changeStore *store.ChangeStore, device *topoca
...
 		return nil, err
 	}
-	log.Println(sync.Device.Addr, "Connected over gNMI")
+	log.Info(sync.Device.Addr, "Connected over gNMI")
 
 	// Get the device capabilities
 	capResponse, capErr := target.CapabilitiesWithString(context, "")
 	if capErr != nil {
-		log.Println(sync.Device.Addr, "Capabilities", err)
+		log.Warning(sync.Device.Addr, "Capabilities", err)
```
Comment (Andrea-Campanella) [inline pkg/southbound/synchronizer/synchronizer.go]: err ?

File: pkg/southbound/synchronizer/synchronizer.go [hunk 17]
```diff
@@ -82,28 +82,28 @@ func (sync *Synchronizer) syncConfigEventsToDevice() {
...
 		if err != nil {
-			log.Println("Event discarded because change is invalid", err)
+			log.Warning("Event discarded because change is invalid", err)
 			continue
 		}
 		gnmiChange, parseError := change.GnmiChange()
 
 		if parseError != nil {
-			log.Println("Parsing error for Gnmi change ", parseError)
+			log.Warning("Parsing error for Gnmi change ", parseError)
```
Comment (Andrea-Campanella) [inline pkg/southbound/synchronizer/synchronizer.go]: same ?

File: pkg/southbound/synchronizer/synchronizer.go [hunk 18]
```diff
@@ -82,28 +82,28 @@ func (sync *Synchronizer) syncConfigEventsToDevice() {
...
 		target, err := southbound.GetTarget(sync.key)
 		if err != nil {
-			log.Println(err)
+			log.Warning(err)
 			continue
 		}
 		setResponse, err := target.Set(sync.Context, gnmiChange)
 		if err != nil {
-			log.Println("SetResponse ", err)
+			log.Warning("SetResponse ", err)
```
Comment (Andrea-Campanella) [inline pkg/southbound/synchronizer/synchronizer.go]: same

File: pkg/southbound/synchronizer/synchronizer.go [hunk 19]
```diff
@@ -113,7 +113,7 @@ func (sync Synchronizer) syncOperationalState() error {
 	target, err := southbound.GetTarget(sync.key)
 
 	if err != nil {
-		log.Println("Can't find target for key", sync.key)
+		log.Warning("Can't find target for key", sync.key)
```
Comment (Andrea-Campanella) [inline pkg/southbound/synchronizer/synchronizer.go]: I would be strong in using errors in this class

File: pkg/dispatcher/dispatcher.go (hunk 1 above)
Comment (SeanCondon) [reply pkg/dispatcher/dispatcher.go]: done

File: pkg/dispatcher/dispatcher.go (hunk 2 above)
Comment (SeanCondon) [reply pkg/dispatcher/dispatcher.go]: done

File: pkg/manager/manager_test.go (hunk 3 above)
Comment (SeanCondon) [reply pkg/manager/manager_test.go]: changed

File: pkg/northbound/gnmi/subscribe.go (hunk 4 above)
Comment (SeanCondon) [reply pkg/northbound/gnmi/subscribe.go]: changed

File: pkg/southbound/clientManager.go (hunk 5 above)
Comment (SeanCondon) [reply pkg/southbound/clientManager.go]: done

File: pkg/southbound/clientManager.go (hunk 15 above)
Comment (SeanCondon) [reply pkg/southbound/clientManager.go]: changed

File: pkg/southbound/synchronizer/synchronizer.go (hunk 16 above)
Comment (SeanCondon) [reply pkg/southbound/synchronizer/synchronizer.go]: done

File: pkg/southbound/synchronizer/synchronizer.go (hunk 17 above)
Comment (SeanCondon) [reply pkg/southbound/synchronizer/synchronizer.go]: done

File: pkg/southbound/synchronizer/synchronizer.go (hunk 18 above)
Comment (SeanCondon) [reply pkg/southbound/synchronizer/synchronizer.go]: done

File: pkg/southbound/synchronizer/synchronizer.go (hunk 19 above)
Comment (SeanCondon) [reply pkg/southbound/synchronizer/synchronizer.go]: done

File: pkg/manager/setconfig.go (hunk 7 above)
Comment (SeanCondon) [reply pkg/manager/setconfig.go]: It's not really a warning - more of a no-op

File: pkg/manager/setconfig.go (hunk 9 above)
Comment (SeanCondon) [reply pkg/manager/setconfig.go]: done

File: pkg/manager/setconfig.go (hunk 8 above)
Comment (SeanCondon) [reply pkg/manager/setconfig.go]: done

File: pkg/northbound/gnmi/set.go (hunk 11 above)
Comment (SeanCondon) [reply pkg/northbound/gnmi/set.go]: done

File: pkg/northbound/gnmi/set.go (hunk 12 above)
Comment (SeanCondon) [reply pkg/northbound/gnmi/set.go]: done

**Pull request merged**